packages/lib/services/rest/routes/notes.ts

```diff
@@ -245,27 +253,36 @@ async function removeTempFiles(urls: string[]) {
 	}
 }
 
-function replaceImageUrlsByResources(markupLanguage: number, md: string, urls: any, imageSizes: any) {
+function replaceUrlsByResources(markupLanguage: number, md: string, urls: any, imageSizes: any) {
```

Comment:
If I'm not mistaken there's a dependency between `urls` and `imageSizes` - basically you get the size of `urls[i]` by getting `imageSizes[i]`. Now that this function is used to process PDF files too, are you sure that these two variables are still in sync? i.e. that we don't end up with the size of `urls[i]` being on `imageSizes[i - 1]`

Reply:
No because `imageSizes` is an object and `i` (i.e the key) here is the url string itself and not index.
```
const imageSizesIndexes: any = {};
```
```
imageSizesIndexes[urlInfo.originalUrl]
```

Reply:
Oh that's right, so no problem then. Thanks for confirming.